### PR TITLE
DOM tree

### PR DESCRIPTION
**Pull request by Beroozam**

### CODE REVIEW

**Comment by mahdiHash** [comment]
Hi @Beroozam.  
Please open a pull request when the translation is ready.  
Since the article and other files are not fully translated yet, I'll convert this PR to draft for now.  
You can set it to "ready for review" whenever you're done translating.  
Good luck!

**Comment by Beroozam** [comment]
Thank you in advance for your guidance

File: 2-ui/1-document/02-dom-nodes/article.md [hunk 1]
```diff
@@ -188,95 +189,95 @@ For example, comments:
...
-Now the last selected element is available as `$0`, the previously selected is `$1` etc.
+اکنون آخرین عنصر انتخاب شده به عنوان `0$` در دسترس است، عنصر انتخاب شده قبلی `1$` و غیره است.
 
-We can run commands on them. For instance, `$0.style.background = 'red'` makes the selected list item red, like this:
+ما می‌توانیم دستورات را روی آنها اجرا کنیم. به عنوان مثال <span dir="ltr">`$0.style.background = 'red'`</span>، آیتمِ لیست انتخاب شده را قرمز می‌کند، مانند این:
 
 ![](domconsole0.svg)
 
-That's how to get a node from Elements in Console.
+به این ترتیب می‌توان یک گره از Elements را از کنسول دریافت کرد.
```
Comment (mahdiHash) [inline 2-ui/1-document/02-dom-nodes/article.md:259]:
```suggestion
به این ترتیب می‌توان یک گره از Elements را در کنسول دریافت کرد.
```

File: 2-ui/1-document/02-dom-nodes/article.md [hunk 2]
```diff
@@ -188,95 +189,95 @@ For example, comments:
...
-There's also a road back. If there's a variable referencing a DOM node, then we can use the command `inspect(node)` in Console to see it in the Elements pane.
+یک راه برگشت هم هست اگر متغیری وجود دارد که به یک گره DOM ارجاع می‌دهد، می‌توانیم از دستور `inspect(node)` در کنسول استفاده کنیم تا آن را در قسمت Elements ببینیم.
 
-Or we can just output the DOM node in the console and explore "in-place", like `document.body` below:
+یا فقط می‌توانیم گره DOM را در کنسول خروجی بگیریم و "درجا" کاوش کنیم، مانند `document.body` در زیر:
 
 ![](domconsole1.svg)
 
-That's for debugging purposes of course. From the next chapter on we'll access and modify DOM using JavaScript.
+این البته برای اهداف اشکال زدایی است. از فصل بعدی به DOM با استفاده از جاوا اسکریپت دسترسی پیدا می‌کنیم و آن را اصلاح می‌کنیم.
```
Comment (mahdiHash) [inline 2-ui/1-document/02-dom-nodes/article.md:267]:
```suggestion
این البته برای اهداف رفع خطا است. از فصل بعدی با استفاده از جاوااسکریپت به DOM دسترسی پیدا می‌کنیم و آن را اصلاح می‌کنیم.
```

File: 2-ui/1-document/02-dom-nodes/article.md [hunk 3]
```diff
@@ -188,95 +189,95 @@ For example, comments:
...
 
-The browser developer tools are a great help in development: we can explore the DOM, try things and see what goes wrong.
+ابزارهای توسعه دهنده مرورگر کمک بزرگی در توسعه هستند: ما می‌توانیم DOM را کاوش کنیم، چیزهایی را امتحان کنیم و ببینیم چه مشکلی پیش می‌آید.
 
-## Summary
+## خلاصه مطلب
 
-An HTML/XML document is represented inside the browser as the DOM tree.
+یک سند HTML/XML در داخل مرورگر به عنوان درخت DOM نشان داده می‌شود.
```
Comment (mahdiHash) [inline 2-ui/1-document/02-dom-nodes/article.md:273]:
```suggestion
یک سند HTML/XML داخل مرورگر به عنوان درخت DOM نشان داده می‌شود.
```

File: 2-ui/1-document/02-dom-nodes/article.md [hunk 4]
```diff
@@ -188,95 +189,95 @@ For example, comments:
...
 
-- Tags become element nodes and form the structure.
-- Text becomes text nodes.
-- ...etc, everything in HTML has its place in DOM, even comments.
+- تگ‌ها به گره‌های عنصر تبدیل می‌شوند و ساختار را تشکیل می‌دهند.
+- متن به گره‌های متنی تبدیل می‌شود.
+- ... و غیره، همه چیز در HTML جایگاه خود را در DOM دارد، حتی نظرات.
```
Comment (mahdiHash) [inline 2-ui/1-document/02-dom-nodes/article.md:277]:
```suggestion
- ... و غیره، همه چیز در HTML جایگاه خود را در DOM دارد، حتی یادداشت‌ها.
```

File: 2-ui/1-document/02-dom-nodes/article.md [hunk 5]
```diff
@@ -188,95 +189,95 @@ For example, comments:
...
 
-We can use developer tools to inspect DOM and modify it manually.
+ما می‌توانیم از ابزارهای توسعه دهنده برای DOM Inspect و اصلاح آن به صورت دستی استفاده کنیم.
```
Comment (mahdiHash) [inline 2-ui/1-document/02-dom-nodes/article.md:279]:
```suggestion
ما می‌توانیم از ابزارهای توسعه دهنده برای بازرسی DOM و اصلاح آن به صورت دستی استفاده کنیم.
```

File: 2-ui/1-document/02-dom-nodes/article.md [hunk 6]
```diff
@@ -188,95 +189,95 @@ For example, comments:
...
 
-Here we covered the basics, the most used and important actions to start with. There's an extensive documentation about Chrome Developer Tools at <https://developers.google.com/web/tools/chrome-devtools>. The best way to learn the tools is to click here and there, read menus: most options are obvious. Later, when you know them in general, read the docs and pick up the rest.
+در اینجا به اصول اولیه، پرکاربردترین و مهم‌ترین اقدامات برای شروع پرداختیم. در <https://developers.google.com/web/tools/chrome-devtools> اسناد گسترده‌ای در مورد ابزارهای برنامه نویس Chrome وجود دارد. بهترین راه برای یادگیری ابزارها این است که آن‌ها امتحان کنید، منوها را بخوانید: اکثر گزینه‌ها واضح هستند. بعداً، وقتی آنها را به طور کلی شناختید، اسناد را بخوانید و بقیه را انتخاب کنید.
 
-DOM nodes have properties and methods that allow us to travel between them, modify them, move around the page, and more. We'll get down to them in the next chapters.
+گره‌های DOM دارای ویژگی‌ها و روش‌هایی هستند که به ما امکان می‌دهند بین آنها پیمایش کنیم، آنها را اصلاح کنیم، در صفحه حرکت کنیم و موارد دیگر. در فصل‌های بعدی به آنها خواهیم پرداخت.
```
Comment (mahdiHash) [inline 2-ui/1-document/02-dom-nodes/article.md:283]:
```suggestion
گره‌های DOM دارای ویژگی‌ها و روش‌هایی هستند که به ما امکان می‌دهند بین آن‌ها پیمایش کنیم، آن‌ها را اصلاح کنیم، در صفحه حرکت کنیم و موارد دیگر. در فصل‌های بعدی به آن‌ها خواهیم پرداخت.
```

**Review by mahdiHash — CHANGES_REQUESTED** [review]
Please review the suggestions.
You can add the suggestions to batch and create a commit just once ([docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes)).

File: 2-ui/1-document/02-dom-nodes/article.md (hunk 2 above)
Comment (Beroozam) [reply 2-ui/1-document/02-dom-nodes/article.md:267]: I think اشکال زدایی is better but رفع خطا also good.  خطا is better for 'error' not for 'debug' @mahdiHash

File: 2-ui/1-document/02-dom-nodes/article.md [hunk 7]
```diff
@@ -142,16 +142,17 @@ let node4 = {"name":"HTML","nodeType":1,"children":[{"name":"HEAD","nodeType":1,
...
-```html no-beautify
-<table id="table"><tr><td>1</td></tr></table>
-```
+برای HTML:
 
-DOM-structure will be:
+ ```html no-beautify
+ <table id="table"><tr><td>1</td></tr></table>
+ ```
+ساختار DOM اینطور خواهد بود:
```
[the remark this answers is not in the document]
Comment (Beroozam) [reply 2-ui/1-document/02-dom-nodes/article.md:155]: I'm sorry for adding a new line

File: 2-ui/1-document/02-dom-nodes/article.md (hunk 4 above)
Comment (Beroozam) [reply 2-ui/1-document/02-dom-nodes/article.md:278]: maybe کامنت is better

File: 2-ui/1-document/02-dom-nodes/article.md [hunk 8]
```diff
@@ -160,26 +161,26 @@ let node5 = {"name":"TABLE","nodeType":1,"children":[{"name":"TBODY","nodeType":
...
 ````
 
-## Other node types
+## انواع دیگر گره‌ها
 
-There are some other node types besides elements and text nodes.
+علاوه بر عناصر و گره‌های متن، انواع گره‌های دیگری نیز وجود دارد.
 
-For example, comments:
+به عنوان مثال، نظرات:
```
[the remark this answers is not in the document]
Comment (mahdyar) [reply 2-ui/1-document/02-dom-nodes/article.md:172]: IMO the phrase "comment" itself is more comprehensible for Persian readers:
```suggestion
به عنوان مثال، کامنت‌ها:
```

Although for the integration of the translation, "یادداشت" can be used as well.

File: 2-ui/1-document/02-dom-nodes/article.md [hunk 9]
```diff
@@ -188,95 +189,95 @@ For example, comments:
...
+12 نوع گره وجود دارد. در عمل ما معمولا با 4 مورد از آنها کار می‌کنیم [12 node types](https://dom.spec.whatwg.org/#node) :
 
-1. `document` -- the "entry point" into DOM.
-2. element nodes -- HTML-tags, the tree building blocks.
-3. text nodes -- contain text.
-4. comments -- sometimes we can put information there, it won't be shown, but JS can read it from the DOM.
+1. `document` -- <span dir="rtl">"نقطه‌ی ورود" به DOM</span>
+2. گره‌های عنصر -- <span dir="rtl">تگ های HTML، بلوک‌های ساختار درختی.</span>
+3. حاوی متن -- گره‌های متن
+4. (comments) -- <span dir="rtl">گاهی اوقات می‌توانیم اطلاعات را در آنجا قرار دهیم، نشان داده نمی‌شود، اما JS می‌تواند آن را از DOM بخواند.</span>
```
[the remark this answers is not in the document]
Comment (mahdiHash) [reply 2-ui/1-document/02-dom-nodes/article.md:213]: For the article, other than code sections, the text will be using RTL when it's ready to publish. You can see this in the [Arabic](https://ar.javascript.info/) JavaScript tutorial.

To translate comments of a code block, you can read [ترجمۀ کامنت‌ها در کد](https://github.com/javascript-tutorial/fa.javascript.info#%D8%AA%D8%B1%D8%AC%D9%85%DB%80-%DA%A9%D8%A7%D9%85%D9%86%D8%AA%D9%87%D8%A7-%D8%AF%D8%B1-%DA%A9%D8%AF) of README. The whole point is to invert the order of the words (if the comment contains English words after translation).

But some English expressions need to be wrapped inside HTML tags with LTR. Most of these expressions start with numbers or symbols. There is an expression in this article which starts with a number followed by English words and you've put that inside a `<span>` tag with LTR. I thought it's not needed but when I checked the Arabic tutorial, I changed my mind. 

Thank you for caring about readability of the contents!

File: 2-ui/1-document/02-dom-nodes/article.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (mahdiHash) [reply 2-ui/1-document/02-dom-nodes/article.md]: No need to be sorry! It just happens :)

File: 2-ui/1-document/02-dom-nodes/article.md (hunk 4 above)
Comment (mahdiHash) [reply 2-ui/1-document/02-dom-nodes/article.md:278]: Yeah, I think "کامنت" is better.
Please change other lines that have "comment" in them, too.

File: 2-ui/1-document/02-dom-nodes/article.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (Beroozam) [reply 2-ui/1-document/02-dom-nodes/article.md:213]:
```suggestion
1. `document` -- "DOM" نقطه‌ی ورود به
2. (گره‌های عنصر) -- بلوک‌های ساختار درختی , HTML تگ‌های 
3. حاوی متن -- (گره‌های متن)
4. (کامنت‌ها) -- بخواند DOM گاهی اوقات می‌توانیم اطلاعات را در آنجا قرار دهیم، نشان داده نمی‌شود، اما جاوااسکریپت می‌تواند آن را از
```

Thank you for your attention and that I use the recommended way for Persian sentences in README and use from span LTR for some of English expression. thank you very much

**Review by Beroozam — COMMENTED** [review]
Finish review

**Review by mahdiHash — APPROVED** [review]
Thanks for your contribution, @Beroozam.

**Review by mahdyar — APPROVED** [review]
LGTM!

**Pull request merged**